src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -875,23 +886,25 @@ public ImmutableArray<T> RemoveAll(Predicate<T> match)
             Requires.NotNull(match, "match");
 
             if (self.IsEmpty)
-            {
-                return new ImmutableArray<T>(self.array);
-            }
+                return self;
```

Comment:
There are a few of these changes in this file (returning an existing instance of the struct rather than creating a new one). Is this a micro-optimization where we avoid creating/copying a struct one additional time? If so, I don't object. This makes it slightly easier to read and (I guess) slightly faster. But that probably needs a new, dedicated pull request to get away from several of the more questionable changes in this PR.